src/functions/Pester.SafeCommands.ps1

```diff
@@ -78,11 +78,15 @@ $script:SafeCommands = @{
     'Write-Warning'        = & $Get_Command -Name Write-Warning        -Module Microsoft.PowerShell.Utility    @safeCommandLookupParameters
 }
 
-# Not all platforms have Get-WmiObject (Nano or PSCore 6.0.0-beta.x on Linux)
-# Get-CimInstance is preferred, but we can use Get-WmiObject if it exists
-# Moreover, it shouldn't really be fatal if neither of those cmdlets
-# exist
-if (($cim = & $Get_Command -Name Get-CimInstance -Module CimCmdlets -CommandType Cmdlet -ErrorAction Ignore)) {
+# Not all platforms have Get-CimInstance (preferred) or Get-WmiObject.
+# It shouldn't be fatal if neither of those cmdlets exists, however
+# some NanoServer/PS images contain a non-functioning version of Get-CimInstance
+# so don't even try it if we're on one of those images.
+$NanoServerRegistryKey = 'HKLM:\Software\Microsoft\Windows NT\CurrentVersion\Server\ServerLevels'
+$NanoServerRegistryKeyValue = (Get-ItemPropertyValue $NanoServerRegistryKey -Name NanoServer -ErrorAction Ignore)
+$NanoServer = $NanoServerRegistryKeyValue -eq 1
```

Comment:
```suggestion
$NanoServer = (& $SafeCommands['Get-ItemProperty'] -Path $NanoServerRegistryKey -ErrorAction Ignore) | & $SafeCommands['ForEach-Object'] -MemberName NanoServer) -eq 1
```

No idea why it's failing on the Windows builds, but `Get-ItemPropertyValue` was introduced in PSv5 so we need to replace that. Maybe try if this works?

We should also call the built-in cmdlets by using the `$SafeCommands`-dictionary (nit).

Reply:
Thank you for that! There's a missing (or extra) brace there, but it seems to work.